### DISCUSSION TITLE
치트 코드에 대한 문의

### DISCUSSION THREAD

**Discussion by mille-feu** [Tiny-Metro/Poly-Metro#200] [Q&A]
문의 사항이 추가될 경우, 지속적으로 추가하도록 하겠습니다.

- 2023.05.21 11:41AM 추가
  - add_station 명령어의 랜덤 위치는 기존 Station 생성 로직 (외접원 관련)을 무시한 랜덤 위치인가요?
  - del_lane 명령어로 노선을 삭제할 때, 운행중인 열차는 어떻게 되나요?
    - 즉시 삭제
    - 기존 삭제 로직과 동일 (다음 역 도달 전까지 삭제 지연)
  - investment_success, fail 명령어는 진행중이지 않은 investment에 대해 작동하지 않는 것이 맞나요?
  - repay 명령어는 소지금이 부족할 경우에는 어떻게 작동하나요?

- 2023.05.21 12:39PM 추가
  - del_station 명령어로 노선이 연결된 역을 삭제할 경우의 처리는 어떻게 되나요?

- 2023.05.21 1:18PM 추가
  - repay {id} 명령어의 구현 방식에 대하여, 다음 두 가지 선택지 혹은 그 외의 방식에 대한 결정이 필요합니다
    - 기존 대출 아이템에 ID 부여 (현재는 ID가 부여되어 있지 않음)
    - investment의 경우처럼, 대출 아이템의 순서로 결정

- 2023.05.21 1:40PM 추가
  - add_complain 명령어로 증감하는 불만도는 정책, 이벤트 등의 증감량 변화에 영향을 받나요?

**Comment by cstrnull00** [comment]
* add_station 명령어 실행 시 생성되는 역은 외접원 관련한 제약을 따릅니다.
* del_lane 명령어로 노선 삭제 시에도 운행 중 열차는 기존 삭제 로직을 따릅니다.
* del_station 명령어로 노선이 연결된 역을 삭제할 경우, 해당 역의 정보만 사라지면 될 것 같습니다.
  * 그냥 원래부터 없었던 것처럼 처리해주시면 되겠습니다. (승객도 삭제)
* repay
  * repay의 id는 investment의 경우처럼 대출 아이템의 순서로 결정하면 될 것 같습니다.
  * 소지금 부족할 시
    1.  소지금을 0원으로 설정
    2.  해당 부족분만큼 -
    3.  repay 시 소지금엔 영향 없음
    세 가지 방안을 생각하고 있는데 @seong0929 의 의견 역시 필요할 것 같습니다.
* add_complain 명령어로 증감하는 불만도는 영향을 받지 않는 상수입니다.
* 진행중이지 않은 investment라는 말은? 좀 더 구체적인 설명이 필요합니다.

**Reply by seong0929** [reply]
- 대출을 갚을 시 소지금 부족하다면 시간이 지남에 따라 게임 오버가 되기에 곤란할 수 있기에 repay를 c와 같이 생각을 하고 있었습니다. 
- investment_success, fail 명령어는 진행중이지 않은 investment에 대해 작동하지 않는 것이 맞나요?
  - 투자 미션을 수락한 경우에만 해당 명령어가 작동합니다.
